src/kitovu/sync/filecache.py

```diff
@@ -143,6 +143,8 @@ def discover_changes(self,
         if not local_full_path.exists():
             return FileState.NEW
 
+        if local_full_path not in self._data:
+            self._data[local_full_path] = File(cached_digest='', plugin_name=plugin.NAME)
```

Comment:
mypy fails with `Argument "plugin_name" to "File" has incompatible type "Optional[str]"; expected "str"` here, but an `assert plugin.NAME is not None` above (probably fine to do it even outside of the `if`) will help.

Reply:
Not sure if it's a good idea to use `''` for the cached digest, that makes it look like it's a bug in digest calculation when you don't know about it. What about using `None` instead?

Reply:
Also, please add a newline after this line, as it's unrelated to the rest of the code creating `file` (makes it easier to scan, IMHO).

Reply:
Good idea with setting the cached_digest to `None`. That way it's clear if it from the plugin or not (beacause plugins can only return strings). Additionally I've added a `assert cached_digest is not None` in the `to_dict` method to ensure it won't get saved that way (either download => local_digest or set it to the current if the `remote_digest` and `local_digest` are equal